include/range/v3/utility/variant.hpp

```diff
@@ -67,21 +67,21 @@ namespace ranges
             template<>
             union variant_data<>
             {
-                void move(std::size_t, variant_data &&) const
+                [[noreturn]] void move(std::size_t, variant_data &&) const
                 {
                     RANGES_ASSERT(false);
                 }
-                void copy(std::size_t, variant_data const &) const
+                [[noreturn]] void copy(std::size_t, variant_data const &) const
                 {
                     RANGES_ASSERT(false);
                 }
-                bool equal(std::size_t, variant_data const &) const
+                [[noreturn]] bool equal(std::size_t, variant_data const &) const
                 {
                     RANGES_ASSERT(false);
                     return true;
                 }
```

Comment:
Functions marked with `[[noreturn]]` cannot return anything. One would also need to remove the `return` statements in these functions: 

```c++
[[noreturn]] bool equal(std::size_t, variant_data const &) const
{
    RANGES_ASSERT(false);
}
```